src/utils/validation.ts

```diff
@@ -1,36 +1,43 @@
 import { ObjectSchema, ValidationResult } from '@hapi/joi';
 import { Component, ComponentReference, Prefab } from '../types';
 
-export const validate = <T extends { name: string }>(
+export const validate = <T extends { jsx?: string; name: string }>(
   schema: ObjectSchema,
   item: T,
 ): void => {
   const { error }: ValidationResult = schema.validate(item);
 
+  const type = item.jsx ? 'Component' : 'Prefab';
+
   if (typeof error !== 'undefined') {
-    throw error;
+    throw new Error(`Property: ${error.message} at ${type}: ${item.name}`);
   }
 };
 
-export const findDuplicates = <T extends { name: string }>(list: T[]): void => {
+export const findDuplicates = <T extends { name: string }>(
+  list: T[],
+  type: string,
+): void => {
   list.reduce((acc: string[], { name }: T): string[] => {
```

Comment:
There is no use for this being a `reduce` (instead of ie. `forEach`) because we return `void`.